packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -37,174 +45,237 @@ export const GridRootStyles = styled('div', {
+
+    '--DataGrid-containerBackground': containerBackground,
+    '--DataGrid-pinnedBackground': pinnedBackground,
+    '--DataGrid-rowBorderColor': borderColor,
+
     '--DataGrid-cellOffsetMultiplier': 2,
+    '--DataGrid-width': '0px',
+    '--DataGrid-hasScrollX': '0',
+    '--DataGrid-hasScrollY': '0',
+    '--DataGrid-offsetTop': '0px',
+    '--DataGrid-offsetLeft': '0px',
+    '--DataGrid-scrollbarSize': '10px',
+    '--DataGrid-rowWidth': '0px',
+    '--DataGrid-columnsTotalWidth': '0px',
+    '--DataGrid-leftPinnedWidth': '0px',
+    '--DataGrid-rightPinnedWidth': '0px',
+    '--DataGrid-headerHeight': '0px',
+    '--DataGrid-headersTotalHeight': '0px',
+    '--DataGrid-topContainerHeight': '0px',
+    '--DataGrid-bottomContainerHeight': '0px',
+
     flex: 1,
     boxSizing: 'border-box',
     position: 'relative',
     borderWidth: '1px',
     borderStyle: 'solid',
     borderColor,
     borderRadius: 'var(--unstable_DataGrid-radius)',
-    color: (theme.vars || theme).palette.text.primary,
-    ...theme.typography.body2,
+    overflow: 'hidden',
+    color: (t.vars || t).palette.text.primary,
+    ...t.typography.body2,
     outline: 'none',
     height: '100%',
     display: 'flex',
     minWidth: 0, // See https://github.com/mui/mui-x/issues/8547
     minHeight: 0,
     flexDirection: 'column',
     overflowAnchor: 'none', // Keep the same scrolling position
-    [`&.${gridClasses.autoHeight}`]: {
+    [`&.${c.autoHeight}`]: {
       height: 'auto',
-      [`& .${gridClasses['row--lastVisible']} .${gridClasses.cell}`]: {
-        borderBottomColor: 'transparent',
-      },
     },
-    [`&.${gridClasses.autosizing}`]: {
-      [`& .${gridClasses.columnHeaderTitleContainerContent} > *`]: {
+    [`&.${c.autosizing}`]: {
+      [`& .${c.columnHeaderTitleContainerContent} > *`]: {
         overflow: 'visible !important',
       },
-      [`& .${gridClasses.cell} > *`]: {
+      [`& .${c.cell} > *`]: {
         overflow: 'visible !important',
         whiteSpace: 'nowrap',
       },
     },
-    [`& .${gridClasses['virtualScrollerContent--overflowed']} .${gridClasses['row--lastVisible']} .${gridClasses.cell}`]:
-      {
-        borderBottomColor: 'transparent',
-      },
-    [`& .${gridClasses.columnHeader}, & .${gridClasses.cell}`]: {
+    [`& .${c.columnHeader}, & .${c.cell}`]: {
       WebkitTapHighlightColor: 'transparent',
       lineHeight: null,
       padding: '0 10px',
       boxSizing: 'border-box',
     },
-    [`& .${gridClasses.columnHeader}:focus-within, & .${gridClasses.cell}:focus-within`]: {
+    [`& .${c.columnHeader}:focus-within, & .${c.cell}:focus-within`]: {
       outline: `solid ${
-        theme.vars
-          ? `rgba(${theme.vars.palette.primary.mainChannel} / 0.5)`
-          : alpha(theme.palette.primary.main, 0.5)
+        t.vars
+          ? `rgba(${t.vars.palette.primary.mainChannel} / 0.5)`
+          : alpha(t.palette.primary.main, 0.5)
       } 1px`,
       outlineWidth: 1,
       outlineOffset: -1,
```

Comment:
I spotted a possible improvement (unrelated to this PR) during debugging for https://github.com/mui/mui-x/pull/10059#discussion_r1440656297
```suggestion
      outlineOffset: -2,
```

This will remove the outline clipping traded with one pixel white space which looks better to me tbh.

Before:
![image](https://github.com/mui/mui-x/assets/12609561/c34da6ee-765b-422d-b91f-6f1da4b9cf4d)

After:
![image](https://github.com/mui/mui-x/assets/12609561/9ff0abf4-2b12-48c3-9f99-d12ec6754fbd)

Reply:
LGTM also.

Reply:
Let's do this after merging this PR though